decode.go

```diff
@@ -16,11 +16,15 @@ type DecodeFeedback interface {
 	// is shorter than internal layer variables (HeaderLength, or the like) say it
 	// should be.  It sets packet.Metadata().Truncated.
 	SetTruncated()
+	// SetConsumed should be called for streams to mention that only a part of
+	// the payload has been consumed (currently useful for TCP only).
+	SetConsumed(uint)
```

Comment:
I'm not seeing anywhere this is currently in use... could we hold off on the whole `Consumed` thing until we actually code in a use for it?